### PR TITLE
Add -OrderBy combinator

### PR DESCRIPTION
**Pull request by artyomVB**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add -OrderBy combinator for aggregate functions

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionOrderBy.h [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+    DataTypePtr getReturnType() const override
+    {
+        return nested_func->getReturnType();
+    }
+
+    void create(AggregateDataPtr place) const override
+    {
+        nested_func->create(place);
+    }
```
Comment (CurtizJ) [inline src/AggregateFunctions/AggregateFunctionOrderBy.h:120]: Here you will need to create data for `FunctionOrderBy` itself and for nested function at the offset of `sizeof(FunctionOrderByData)`.

**Review by CurtizJ — COMMENTED** [review]
Also you need to create and register the implementation of `IAggregateFunctionCombinator` to be able to call aggregate function from SQL. ([example](https://github.com/ClickHouse/ClickHouse/blob/master/src/AggregateFunctions/AggregateFunctionDistinct.cpp))

Also you need to implement `serialize`, `deserialize` and `merge` methods (see comments in `IAggregateFunction.h.`) to be able to execute aggregation in distributed queries.

File: src/AggregateFunctions/AggregateFunctionOrderBy.h [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        size_t column_size = reinterpret_cast<const OrderByData *>(place)->data[0]->size();
+        writeBinary(column_size, buf);
+        auto size = reinterpret_cast<const OrderByData *>(place)->data.size();
+        for (size_t i = 0; i != size; ++i) {
+            for (size_t j = 0; j != column_size; ++j) {
+                auto ob_data = reinterpret_cast<const OrderByData *>(place);
+                writeBinary((*ob_data->data[i])[j].get<TypeIndexToType<TypeIndex::UInt8>>(), buf);
+            }
+        }
+    }
```
Comment (CurtizJ) [inline src/AggregateFunctions/AggregateFunctionOrderBy.h:174]: It's better to `serialize/deserialize` the whole column with `{serialize/deserialize}BinaryBulkWithMultipleStreams` methods. You can get serialization object from data type by `getDefaultSerialization` method.

Here are some examples:
https://github.com/ClickHouse/ClickHouse/blob/182ba1c6059da5c1c3314b31b97e8491dd34563f/src/Formats/NativeReader.cpp#L68-L84
https://github.com/ClickHouse/ClickHouse/blob/182ba1c6059da5c1c3314b31b97e8491dd34563f/src/Formats/NativeWriter.cpp#L49-L65

File: programs/server/rocksdb/OPTIONS-000027 [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# This is a RocksDB option file.
```
Comment (CurtizJ) [inline programs/server/rocksdb/OPTIONS-000027:1]: Please, remove all accidentally added files.

**Review by CurtizJ — COMMENTED** [review]
Please, add some functional tests. You can find examples in `tests/queries/0_stateless` directory.

File: src/AggregateFunctions/AggregateFunctionOrderBy.h [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        ans.push_back(curr);
+        return ans;
+    }
+
+    void parseString(const String& s)
+    {
+        auto strs = join(s);
+        for (String& str : strs) {
+            std::transform(str.begin(), str.end(), str.begin(), [](unsigned char c) -> unsigned char { return std::toupper(c); });
+            auto pos = str.find("DESC");
```
Comment (BiteTheDDDDt) [inline src/AggregateFunctions/AggregateFunctionOrderBy.h:60]: Is it possible that our `column name` contain desc?

**Pull request closed without merging**